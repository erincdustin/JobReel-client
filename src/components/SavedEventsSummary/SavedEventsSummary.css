.saved-event-list {
    position: relative;
    grid-column-start: 4;
    grid-column-end: 9;
    height: 100%;
    max-height: 450px;
    border: 1px solid black;
    border-radius: 10px;
    position: relative;
    background: #3C3230;
    color: #eeebd9;
    text-align: center;
    overflow-y: auto;
}

.savedEventFilterControls { 
    width: 90%;
    text-align: right;
}

@media screen and (max-width: 960px) {
    .saved-event-list {
        grid-column-start: 1;
        grid-column-end: 9;
    }
}