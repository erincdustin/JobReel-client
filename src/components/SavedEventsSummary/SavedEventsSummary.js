import React from 'react';
import Button from '../../components/Button/Button';
import JobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';
import SavedEvent from '../SavedEvent/SavedEvent';

class SavedEventsSummary extends React.Component {

  static contextType = JobReelContext;

  handleClickDelete(eventId){
    jobReelApiService.deleteEvent(eventId)
    this.context.deleteEvent(eventId)
  }

  render(){
    let mappedEvents;
    if(this.context.savedEvents !== []){
      let events = this.context.savedEvents
      mappedEvents = events
      .map(event => <SavedEvent id={event.event_id} name={event.event_name} host={event.host} city={event.city} state={event.state} address={event.address} date={event.date} url={event.url} desc={event.description} status={event.status} user={event.user_id}/>)
    }

    return(
      <div className="saved-event-list">
        <h2>Saved Events</h2>
        <h3>Events</h3>
        <Button onClick={() => this.context.setManualEventAdd(true)} type="button">Add Event</Button>
        {mappedEvents}
      </div>
    )
  }
}

export default SavedEventsSummary