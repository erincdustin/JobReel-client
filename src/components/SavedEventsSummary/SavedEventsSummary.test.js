import React from 'react'
import ReactDOM from 'react-dom'
import SavedEventsSummary from './SavedEventsSummary'
import PropTypes from 'prop-types'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


class ContextProvider extends React.Component{
  static childContextTypes = {
    myContext: PropTypes.object,
  }
  
  getChildContext = () => ({
    myContext: {
      savedEvents: [
        {
          event_name: 'name',
          host: 'host',
          city: 'city',
          state: 'MD',
          address: '111 St',
          url: 'http://website.com',
          status: 'Will Attend',
          description: 'description',
          event_id: 1,
          date_added: '2019-08-26T15:03:05.646Z',
          user_id: 1
        }
      ],
      setManualEventAdd: () => { }
    }
  })

  render(){
    return this.props.children
  }
}

describe('SavedEventsSummary smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<SavedEventsSummary />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('SavedEventsSummary snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<ContextProvider><SavedEventsSummary/></ContextProvider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})