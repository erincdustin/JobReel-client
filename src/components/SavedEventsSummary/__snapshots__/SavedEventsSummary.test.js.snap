// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedEventsSummary snapshot test renders the UI as expected 1`] = `<SavedEventsSummary />`;
